RecoLocalTracker/SiPixelRecHits/src/PixelCPEGeneric.cc

```diff
@@ -518,46 +570,147 @@ float PixelCPEGeneric::generic_position_formula(int size,                    //!
-      Q_l_Y += pix_adc;
+    cluster_matrix[1 + pixel.x - xmin][1 + pixel.y - ymin] = pixel.adc;
   }
 
+  // Iterate over the pixels.
+  bool pruned_cluster = false;  //indicating whether any rows/columns have been deleted from the cluster
+  while (!pruned_cluster) {
+    Q_f_X = Q_l_X = 0.0;
+    Q_f_Y = Q_l_Y = 0.0;
+    for (int i = 0; i != isize; ++i) {
+      auto const& pixel = theClusterParam.theCluster->pixel(i);
+      // ggiurgiu@fnal.gov: add pixel charge truncation
+      int pix_adc = pixel.adc;
+      if (UseErrorsFromTemplates_ && TruncatePixelCharge_)
+        pix_adc = std::min(pix_adc, theClusterParam.pixmx);
+
+      if (!correct_xtalk) {  //read the charges exactly as before if correction is off
+        // X projection
+        if (pixel.x == xmin)
+          Q_f_X += pix_adc;
+        if (pixel.x == xmax)
+          Q_l_X += pix_adc;
+        //
+        // Y projection
+        if (pixel.y == ymin)
+          Q_f_Y += pix_adc;
+        if (pixel.y == ymax)
+          Q_l_Y += pix_adc;
+      }
+      // correct the charges modified by xtalk
+      else {
+        if (pixel.x == xmin || pixel.x == xmax) {
+          float delta_charge = 0;
+          int oddness = 2 * (pixel.x % 2) - 1;  // 1 for odd and -1 for even
+          for (int direction = -1; direction < 2; direction += 2) {
+            delta_charge += xtalk_correction_factors[2 + oddness * direction][0] * pix_adc +
+                            xtalk_correction_factors[2 + oddness * direction][1] *
+                                cluster_matrix[1 + pixel.x + oddness - xmin][1 + pixel.y - ymin] -
+                            pix_adc;
+          }
+          if (delta_charge + pix_adc < induced_thre)
+            delta_charge = -1 * pix_adc;
+          if (pixel.x == xmin) {
+            Q_f_X += delta_charge + pix_adc;
+          }
+          if (pixel.x == xmax) {
+            Q_l_X += delta_charge + pix_adc;
+          }
+        }
+        if (pixel.y == ymin || pixel.y == ymax) {
+          float delta_charge = 0;
+          int oddness = 2 * (pixel.y % 2) - 1;  // 1 for odd and -1 for even
+          for (int direction = -1; direction < 2; direction += 2) {
+            delta_charge += xtalk_correction_factors[1 + oddness * direction][0] * pix_adc +
+                            xtalk_correction_factors[2 + oddness * direction][1] *
+                                cluster_matrix[1 + pixel.x - xmin][1 + pixel.y + oddness - ymin] -
+                            pix_adc;
+          }
+          if (delta_charge + pix_adc < induced_thre)
+            delta_charge = -1 * pix_adc;
+          if (pixel.y == ymin)
+            Q_f_Y += delta_charge + pix_adc;
+          if (pixel.y == ymax)
+            Q_l_Y += delta_charge + pix_adc;
+        }
+      }
+    }
+    if (!correct_xtalk)
+      break;
+    if (Q_f_X == 0 || Q_l_X == 0 || Q_f_Y == 0 ||
+        Q_l_Y == 0) {  //basically if after correction a whole row/column is below threshold
```

Comment:
put the comment on a separate line so the code formatting doesn't cause a strange linebreak like this